src/nodejs/lambda-handlers/submission.js

```diff
@@ -301,6 +317,46 @@ async function mapMetadataMethod(event, user) {
   return mappedMetadata;
 }
 
+async function createStepReviewApprovalMethod(event, user) {
+  const { submissionId, stepName, userIds } = event.params;
+  const approvedUserPrivileges = ['ADMIN', 'REQUEST_DAACREAD', 'REQUEST_ADMINREAD', 'CREATE_STEPREVIEW', 'REMOVE_STEPREVIEW'];
+  if (!user.user_privileges.some((privilege) => approvedUserPrivileges.includes(privilege))) {
+    return { error: 'Not Authorized' };
+  }
+  // This line has a length of 104. Maximum allowed is 100 to fix this
+  const subId = submissionId;
+  const ids = userIds;
+  const param = {
+    submission_id: subId,
+    step_name: stepName,
+    user_ids: ids,
+    submitted_by: user.id
+  };
+  const formData = await db.submission.createStepReviewApproval(param);
+  return formData;
+}
+
+async function getStepReviewApprovalMethod(event, user) {
+  const { id } = event.params;
+  const approvedUserPrivileges = ['ADMIN', 'REQUEST_DAACREAD', 'REQUEST_ADMINREAD', 'CREATE_STEPREVIEW', 'REMOVE_STEPREVIEW'];
```

Comment:
Should check  `'REQUEST_DAACREAD', 'REQUEST_READ' and 'ADMIN' `only.  And should only allow users with just REQUEST_READ if they are on the contributors list.